internal/providersdkv2/resource_vault_cluster.go

```diff
@@ -1290,6 +1441,54 @@ func getValidObservabilityConfig(config map[string]interface{}) (*vaultmodels.Ha
 		}
 	}
 
+	if httpURI != "" || httpMethod != "" || httpCodec != "" {
+		if observabilityConfig != nil {
+			return nil, tooManyProvidersErr
+		}
+
+		if httpURI == "" || httpMethod == "" || httpCodec == "" {
+			missingParamErr = diag.Errorf("http configuration is invalid: configuration information missing")
+		}
+
+		var httpBearerAuth *vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth
+		var httpBasicAuth *vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth
+
+		if httpBearerToken != "" {
+			httpBearerAuth = &vaultmodels.HashicorpCloudVault20201125HTTPBearerAuth{
+				Token: httpBearerToken,
+			}
+
+			// only one of basic or bearer authentication should be submitted
+			if httpBasicUser != "" || httpBasicPassword != "" {
+				missingParamErr = diag.Errorf("http configuration is invalid: either the basic or bearer authentication method can be submitted, but not both")
+			}
+		} else {
+			// http basic requires both the username and password to be filled
+			if httpBasicUser != "" && httpBasicPassword == "" || httpBasicUser == "" && httpBasicPassword != "" {
+				missingParamErr = diag.Errorf("http configuration is invalid: basic authentication requires username and password")
+			} else {
+				httpBasicAuth = &vaultmodels.HashicorpCloudVault20201125HTTPBasicAuth{
+					User:     httpBasicUser,
+					Password: httpBasicPassword,
+				}
+			}
+		}
+
+		observabilityConfig = &vaultmodels.HashicorpCloudVault20201125ObservabilityConfig{
+			HTTP: &vaultmodels.HashicorpCloudVault20201125HTTP{
+				Headers:       httpHeaders,
+				Bearer:        httpBearerAuth,
+				Basic:         httpBasicAuth,
+				Codec:         (*vaultmodels.HashicorpCloudVault20201125HTTPEncodingCodec)(&httpCodec),
+				Compression:   httpCompression,
+				PayloadPrefix: httpPayloadPrefix,
+				PayloadSuffix: httpPayloadSuffix,
+				Method:        httpMethod,
+				URI:           httpURI,
+			},
+		}
+	}
+
 	if missingParamErr != nil {
```

Comment:
I wonder if we should change this name now, it's more like `invalidProviderConfigErr` or smt. I really like the usage in this PR tho, it maintains the prioritization of "too many providers" over invalid settings from specific providers